awx/api/serializers.py

```diff
@@ -2350,6 +2350,7 @@ class Meta:
 
 class InventorySourceOptionsSerializer(BaseSerializer):
     credential = DeprecatedCredentialField(help_text=_('Cloud credential to use for inventory updates.'))
+    source = serializers.ChoiceField(choices=CLOUD_PROVIDERS)
```

Comment:
I did some googling, and it looks like the proper way of solving this is not populating the choices at the module import time but pushing it further towards the serializer instance initialization. So the class attribute definition would look like this:

```suggestion
    source = serializers.ChoiceField()
```

and the initializer could be something like

```python
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.fields['source'].choices = discover_available_plugin_names()
```

But if you can implement the idea of putting an attribute onto the settings object, perhaps you could even do

```python
from django.conf import settings
```

and use that as
```python
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.fields['source'].choices = settings.INVENTORY_SOURCE_PLUGIN_NAMES
```

Reply:
Thank you so much for this guidance!! I believe I have successfully implemented this via this commit: https://github.com/ansible/awx/pull/15537/commits/9c2f448b5f01c5ae102e40ee680c690d4c573328

Reply:
Yeah, it needs further integration/refactoring, though. Let's resolve this thread and continue elsewhere. I've posted more comments in other places.